test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1828 @@
+                            self.node_info[self.MOM_LOGS][0]]
+        for dir in dirs_in_snapshot:
+            rel_path = os.path.join(self.outdir, dir)
+            os.makedirs(rel_path, 0755)
+
+    def __grep_file(self, fd_in, fd_out, pattern):
+        """
+        Grep the file given for the pattern specified
+
+        :param fd_in: file object of the file to grep
+        :type fd_in: file
+        :param fd_out: file object of the file to store grep'd contents
+        :type fd_out: file
+        :param pattern: the pattern to grep for
+        :type pattern: String
+        """
+        for line in fd_in:
+            match = re.search(pattern, line)
+            if match:
+                fd_out.write(line)
+
+    def __capture_cmd_output(self, host, out_path, cmd, grep_pbs=False,
+                             skip_anon=False):
+        """
+        Run a command on the host specified and capture its output
+
+        :param host: the host to run the command on
+        :type host: String
+        :param out_path: path of the output file for this command
+        :type out_path: String
+        :param cmd: The command to execute
+        :type cmd: List
+        :param grep_pbs: Should do "grep pbs" on the output of the command?
+        :type grep_pbs: Boolean
+        :param skip_anon: Skip anonymization even though anonymize is True?
+        :type skip_anon: Boolean
+        """
+        self.logger.debug("capturing " + str(cmd))
+
+        if "qmgr" in cmd[0]:
+            # qmgr -c is being called
+            if self.du.is_localhost(host) is False:
+                # For remote hosts, wrap qmgr's command in quotes
+                cmd[2] = "\'" + cmd[2] + "\'"
+
+        with open(out_path, "w") as out_fd:
+            try:
+                cmd_stdout = out_fd
+                if grep_pbs is True:
+                    # Set stdout to a temp file
+                    fd, fn = self.du.mkstemp()
+                    tempfd = os.fdopen(fd, "w+")
+                    cmd_stdout = tempfd
+                self.du.run_cmd(host, cmd=cmd, stdout=cmd_stdout,
+                                sudo=self.sudo)
+                if grep_pbs is True:
+                    # This is done for ps and lsof commands
+                    # Preserve the header/first line of their outputs
+                    out_fd.write(tempfd.readline())
+                    # Grep for "pbs" and write it to the output file
+                    self.__grep_file(tempfd, out_fd, "pbs")
+                    # Close and delete the temp file
+                    tempfd.close()
+                    os.remove(fn)
+            except Exception, e:
+                self.logger.error(str(e))
+
+            if self.anonymize is True and skip_anon is False:
+                self.__anonymize_file(out_path)
+
+        if self.create_tar is True:
+            self.__add_to_archive(out_path)
+
+    def __initialize_server(self, hostname=None, diag=None):
```

Comment:
You don't need this method as all this operations are handled by PBSService() class itself.